jupyterhub_moss/spawner.py

```diff
@@ -276,140 +295,129 @@ async def create_options_form(spawner):
             jsondata=jsondata,
         )
 
-    # Options retrieved from HTML form and associated converter functions
-    _FORM_FIELD_CONVERSIONS = {
-        "partition": str,
-        "runtime": str,
-        "nprocs": int,
-        "mem": str,
-        "reservation": str,
-        "ngpus": int,
-        "options": lambda v: v.strip(),
-        "output": lambda v: v == "true",
-        "environment_path": str,
-        "default_url": str,
-        "root_dir": str,
-    }
-
-    def __convert_formdata(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
+    def __convert_formdata(self, formdata: dict[str, list[str]]) -> FormOptions:
         """Convert expected input to appropriate type"""
-        options = {}
-        for name, convert in self._FORM_FIELD_CONVERSIONS.items():
-            if name not in formdata:
-                continue
-            value = formdata[name][0].strip()
-            if len(value) == 0:
-                continue
+
+        def get_from_formdata(name: str, default: str = "") -> str:
             try:
-                options[name] = convert(value)
+                return formdata.get(name, (default,))[0].strip()
             except ValueError:
                 raise RuntimeError(f"Invalid {name} value")
 
-        return options
+        # Convert output option from boolean to file pattern
+        has_output = get_from_formdata("output") == "true"
+        output = "slurm-%j.out" if has_output else "/dev/null"
+
+        return dict(
```

Comment:
Typing requests keys to be passed explicitly, so changing the conversion code from using a dict of converters to this.